README.md
yiipin 1.3 crack
======

yiipin is like "Pinterest" 's system with yii1.x. It is a commercial and close source systems, these code is for learning. 



changelog.md
# YiiPin Change Log

## v1.0（20121010）
1. 第一个版本发布

## v1.1（20121101）
1. （新增）增加了淘宝账号登录
1. （新增）增加了好店频道
2. （增强）所有瀑布流页面自动输出第一屏，更利于搜索引擎收录
3. （修正）修正了首页标签合成图按照分类筛选商品的逻辑（感谢型男网）
4. （增强）用户登录、退出增加了跳转页面，以输出Ucenter同步JS代码，同步其他挂接Ucenter的系统如discuz!等的用户状态。
5. （修正）修正用户主页左侧加关注按钮无法使用的BUG（感谢型男网）
6. （修正）修正商品图片上传的类型验证问题
7. （修正）修正后台管理商品图片缩略图处理时对扩展名判断不严格的问题
8. （增强）大部分需要批量操作的后台管理模块增加了批量删除功能
9. （增强）商品分享改用百度分享，支持同时分享商品图片到QQ空间、腾讯微博、新浪微博等
10. （新增）增加了导航栏管理功能
11. （修正）修正默认模板翻页栏当前页面未高亮显示的BUG（感谢型男网）
12. （修正）经过实验确定暂时不支持PHP5.3
13. （增强）增强了瀑布流的服务器端和客户端执行效率
14. （增强）首页增加了可以打开关闭的新商品瀑布流展示功能
15. （增强）淘宝客采集功能增加了卖家信誉度、消保、如实描述、价格范围、佣金比例和商城商品限定选项
16. （修正）淘宝客采集增加图片类型验证机制，过滤无效图片地址的商品
17. （新增）增加了淘宝客采集的商品的批量维护功能，包括下架商品标记、更新名称和价格、店铺关联等
18. （新增）当商品已下架，将在详细页面提示，并提供搜索推广链接和店铺推广链接，不流失流量
19. （新增）后台新增淘宝店铺采集功能，采集来的店铺将自动和已有分享的商品关联起来
20. （新增）后台增加了全站维护关闭服务开关
21. （增强）图片分享（通过拾宝工具抓取的）增加了原页面外链
22. （修正）修正了分享图片时标题截取没有限定长度的问题（感谢型男网）
23. （新增）增加了nginx配置示例文件
24. （增强）商品分享详细页面的大图高度随宽度自动计算，兼容个别很高的图片

## v1.2（20121203）
1. （增强）当商品名称自定义编辑过后，商品维护时不再更新其名称
2. （新增）支持又拍云和阿里云存储图片
3. （新增）支持淘宝商品评论自动采集功能
4. （增强）商品图片抓取加入自动重试机制
5. （新增）增加了简单的积分功能和设置
6. （增强）淘宝客采集增加商品排序设置
7. （增强）淘宝客采集增加分类限定
7. （新增）淘宝商品支持多图显示
8. （修正）首页标签合成图不能合成子类图片的问题
9. （增强）淘宝商品评论采集支持自动注册马甲
10. (新增) 增加了积分兑换频道，支持虚拟、实物积分商品兑换
11. （修改）将宝贝瀑布流默认排序改为最新倒序排列
12. （增强）默认模板顶部增加淘宝登录按钮

## v1.3（20130101）
1. （新增）试用频道
2. （变更）支持淘宝远程图片，同时支持其他网站和图片分享本地图片（商品维护后将自动变更为远程淘宝图片，删除本地淘宝图片）
3. （取消）删除云存储支持（徒增复杂度，很少用到）
3. （新增）调整系统架构以支持主题更换（主题后续奉上）
3. （增强）淘宝客商品采集增加设置记忆功能
4. （增强）淘宝客商品采集增加关键词排除功能
5. （新增）安装包增加IIS7.0干净URL配置文件
6. （修正）UCenter中已有用户同步登录到系统无法自动注册的BUG
7. （增强）前台分享商品增加重复商品检测，如果存在则自动分享已有商品
8. （增强）前台分享商品增加30秒内重复分享屏蔽
9. （增强）后台增加新浪微博、QQ登录验证META设置
10. （新增）淘宝商品采集增加店铺搜索采集功能，不受API权限限制，并结合好店展示
11. （增强）后台淘宝客采集增加总成交量限制条件
12. （增强）后台淘宝客采集增加记录商品成交量、推广量、佣金比例和佣金信息
13. （增强）后台淘宝客采集流程、稳定性优化，改为先分析数据，再商品入库、图片下载
14. （增强）后台淘宝客采集增加中断继续功能
15. （修正）修正淘宝评论采集的一个BUG，增强稳定性
16. （增强）后台淘宝客采集增加API调用超限错误提示
17. （增强）后台增加URL后缀设置，如.html
18. （增强）淘宝评论采集增加关键词过滤功能
19. （增强）淘宝商品主图如果下载失败再次被浏览时重试下载
20. （新增）淘宝商品采集新增天猫精品库采集功能
21. （修正）焦点图幻灯显示最后一张图不显示的BUG
22. （增强）后台增加分享管理
23. （增强）第三方登录现在可选自动注册用户了，更加方便
24. （增强）好店增加店铺品牌和品牌介绍属性，并显示在店铺、商品分享页面
25. （增强）宝贝外链采用JS加密跳转，更利于SEO
26. （增强）逛宝贝和分类页面标题增加标签名称，更利于SEO
27. （优化）宝贝列表页面等大幅度降低数据库查询数量，提高运行速度
28. （增强）后台商品管理列表改为100项每页，增加批量分类功能
29. （增强）后台商品管理增加淘客佣金比例、30天推广量和成交量显示，便于筛选管理
30. （优化）商品列表页面默认排序按照淘客30天内成交量倒序排列
31. 

js/base64.js
function utf16to8(str) {
    var out, i, len, c;

    out = "";
    len = str.length;
    for(i = 0; i < len; i++) {
        c = str.charCodeAt(i);
        if ((c >= 0x0001) && (c <= 0x007F)) {
            out += str.charAt(i);
        } else if (c > 0x07FF) {
            out += String.fromCharCode(0xE0 | ((c >> 12) & 0x0F));
            out += String.fromCharCode(0x80 | ((c >>  6) & 0x3F));
            out += String.fromCharCode(0x80 | ((c >>  0) & 0x3F));
        } else {
            out += String.fromCharCode(0xC0 | ((c >>  6) & 0x1F));
            out += String.fromCharCode(0x80 | ((c >>  0) & 0x3F));
        }
    }
    return out;
}
function utf8to16(str) {
    var out, i, len, c;
    var char2, char3;

    out = "";
    len = str.length;
    i = 0;
    while(i < len) {
        c = str.charCodeAt(i++);
        switch(c >> 4)
        {
          case 0: case 1: case 2: case 3: case 4: case 5: case 6: case 7:
            // 0xxxxxxx
            out += str.charAt(i-1);
            break;
          case 12: case 13:
            // 110x xxxx   10xx xxxx
            char2 = str.charCodeAt(i++);
            out += String.fromCharCode(((c & 0x1F) << 6) | (char2 & 0x3F));
            break;
          case 14:
            // 1110 xxxx  10xx xxxx  10xx xxxx
            char2 = str.charCodeAt(i++);
            char3 = str.charCodeAt(i++);
            out += String.fromCharCode(((c & 0x0F) << 12) |
                                           ((char2 & 0x3F) << 6) |
                                           ((char3 & 0x3F) << 0));
            break;
        }
    }

    return out;
}

var base64EncodeChars = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";
var base64DecodeChars = new Array(
    -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1,
    -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1,
    -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, 62, -1, -1, -1, 63,
    52, 53, 54, 55, 56, 57, 58, 59, 60, 61, -1, -1, -1, -1, -1, -1,
    -1,  0,  1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11, 12, 13, 14,
    15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, -1, -1, -1, -1, -1,
    -1, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40,
    41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, -1, -1, -1, -1, -1);

function base64encode(str) {
    var out, i, len;
    var c1, c2, c3;

    len = str.length;
    i = 0;
    out = "";
    while(i < len) {
        c1 = str.charCodeAt(i++) & 0xff;
        if(i == len)
        {
            out += base64EncodeChars.charAt(c1 >> 2);
            out += base64EncodeChars.charAt((c1 & 0x3) << 4);
            out += "==";
            break;
        }
        c2 = str.charCodeAt(i++);
        if(i == len)
        {
            out += base64EncodeChars.charAt(c1 >> 2);
            out += base64EncodeChars.charAt(((c1 & 0x3)<< 4) | ((c2 & 0xF0) >> 4));
            out += base64EncodeChars.charAt((c2 & 0xF) << 2);
            out += "=";
            break;
        }
        c3 = str.charCodeAt(i++);
        out += base64EncodeChars.charAt(c1 >> 2);
        out += base64EncodeChars.charAt(((c1 & 0x3)<< 4) | ((c2 & 0xF0) >> 4));
        out += base64EncodeChars.charAt(((c2 & 0xF) << 2) | ((c3 & 0xC0) >>6));
        out += base64EncodeChars.charAt(c3 & 0x3F);
    }
    return out;
}

function base64decode(str) {
    var c1, c2, c3, c4;
    var i, len, out;

    len = str.length;
    i = 0;
    out = "";
    while(i < len) {
        /* c1 */
        do {
            c1 = base64DecodeChars[str.charCodeAt(i++) & 0xff];
        } while(i < len && c1 == -1);
        if(c1 == -1)
            break;

        /* c2 */
        do {
            c2 = base64DecodeChars[str.charCodeAt(i++) & 0xff];
        } while(i < len && c2 == -1);
        if(c2 == -1)
            break;

        out += String.fromCharCode((c1 << 2) | ((c2 & 0x30) >> 4));

        /* c3 */
        do {
            c3 = str.charCodeAt(i++) & 0xff;
            if(c3 == 61)
                return out;
            c3 = base64DecodeChars[c3];
        } while(i < len && c3 == -1);
        if(c3 == -1)
            break;

        out += String.fromCharCode(((c2 & 0XF) << 4) | ((c3 & 0x3C) >> 2));

        /* c4 */
        do {
            c4 = str.charCodeAt(i++) & 0xff;
            if(c4 == 61)
                return out;
            c4 = base64DecodeChars[c4];
        } while(i < len && c4 == -1);
        if(c4 == -1)
            break;
        out += String.fromCharCode(((c3 & 0x03) << 6) | c4);
    }
    return out;
}
//js_base64 加密
function str_encode(str){
    return base64encode(utf16to8(str));
}
//js_base64 解密
function str_decode(str){
        return utf8to16(base64decode(str));
}
